tests/test_job_management.py

```diff
@@ -2639,6 +2640,383 @@ def test_inspect_status(dispatcher_live_fixture, request_cred, roles, include_se
+@pytest.mark.parametrize("pass_job_id", [True, False])
+def test_inspect_jobs(dispatcher_live_fixture, request_cred, roles, pass_job_id):
+    required_roles = ['job manager']
+    DispatcherJobState.remove_scratch_folders()
+    server = dispatcher_live_fixture
+    logger.info("constructed server: %s", server)
+    token_none = (request_cred == 'public')
+
+    token_payload = {**default_token_payload}
+
+    if token_none:
+        encoded_token = None
+    else:
+        # let's generate a valid token
+        token_payload["roles"] = roles
+        encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    params = {
+        'query_status': 'new',
+        'product_type': 'dummy',
+        'query_type': "Dummy",
+        'instrument': 'empty',
+        'token': encoded_token,
+    }
+
+    # just for having the roles in a list
+    roles = roles.split(',')
+    roles[:] = [r.strip() for r in roles]
+
+    jdata_done = ask(server,
+                params,
+                expected_query_status=["done"],
+                max_time_s=150,
+                )
+
+    params = {
+        'query_status': 'new',
+        'product_type': 'failing',
+        'query_type': "Dummy",
+        'instrument': 'empty',
+        'token': encoded_token,
+    }
+
+    jdata_failed = ask(server,
+                       params,
+                       expected_query_status='failed'
+                       )
+
+    job_id_done = jdata_done['job_monitor']['job_id']
+    session_id_done = jdata_done['session_id']
+    job_id_failed = jdata_failed['job_monitor']['job_id']
+    session_id_failed = jdata_failed['session_id']
+
+    # generate a new valid token with the same approach
+    if token_none:
+        encoded_token = None
+    else:
+        # let's generate a valid token
+        token_payload["roles"] = roles
+        token_payload["exp"] = int(time.time()) + 5000
+        encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    status_code = 403
+    error_message = ''
+    if request_cred == 'invalid_token':
+        # an invalid (encoded) token, just a string
+        encoded_token = 'invalid_token'
+        error_message = 'The token provided is not valid.'
+    elif request_cred == 'public':
+        error_message = 'A token must be provided.'
+    elif request_cred == 'private':
+        if 'job manager' not in roles:
+            lacking_roles = ", ".join(sorted(list(set(required_roles) - set(roles))))
+            error_message = (
+                f'Unfortunately, your privileges are not sufficient to inspect the state for a given job_id.\n'
+                f'Your privilege roles include {roles}, but the following roles are missing: {lacking_roles}.'
+            )
+
+    inspect_params = dict(
+                         token=encoded_token
+                     )
+    if pass_job_id:
+        inspect_params['job_id'] = job_id_done[:8]
+
+    c = requests.get(server + "/inspect-jobs",
```

Comment:
I am concerned this will become too slow becasue of all the additional reads. Could you check timing here, and maybe also number of openned files.

Reply:
indeed there were several opened files that were just read but the content was not included within the output

I reformatted the code so that the two functionalities are now separated

Reply:
Yes, but check how long it takes in the test. And I suspect you might be able to trace how many files were opened.

Reply:
execution times I read (in the tests) range from 0.003 sec to 0.006 sec

Reply:
Yeah it's hard to check timing in test. 
For the files, can you find a way?

Reply:
22 files where read when running the `test_inspect_jobs_with_callbacks` test
3 files where read when running `test_inspect_jobs_failed` test

for each inspect-jobs call the following files are opened (and closed):
- `query_output.json`
- `job_monitor*.json` (depending on the number of nodes per job)
- `analysis_parameters.json`

Reply:
if the output of inspect-state is to be included within the output of inspect-jobs, than also the following files are open:
- `session.log`

Reply:
I indirectly do it by inspecting the content of the returned object.

What would you like to see?

Reply:
> if the output of inspect-state is to be included within the output of inspect-jobs, than also the following files are open:
> 
>     * `session.log`

with the args `exclude_analysis_parameters` the content of the `analysis_parameters.json` file can be excluded thusr educing the number of files opened